dvc/project.py

```diff
@@ -1165,22 +1158,32 @@ def metrics_add(self, path, typ=None, xpath=None):
     def metrics_remove(self, path):
         self._metrics_modify(path, delete=True)
 
-    def graph(self, from_directory=None):
+    def graph(self, stages=None, from_directory=None, cwd=None):
         import networkx as nx
-        from dvc.exceptions import OutputDuplicationError
+        from dvc.exceptions import (OutputDuplicationError,
+                                    WorkingDirectoryAsOutputError)
 
         G = nx.DiGraph()
         G_active = nx.DiGraph()
-        stages = self.stages(from_directory)
-
+        stages = stages or self.stages(from_directory)
         outs = []
         outs_by_path = {}
+
+        if cwd:
```

Comment:
Let's check cwd for every stage and not only for the one that is being created. Same as we check output duplication for all stages, and not only for the new one.